http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/HttpCromwellDAO.scala

```diff
@@ -0,0 +1,91 @@
+package org.broadinstitute.dsde.workbench.leonardo.dao
+
+import akka.http.scaladsl.model.StatusCode
+import cats.effect.Async
+import cats.mtl.Ask
+import cats.syntax.all._
+import io.circe._
+import org.broadinstitute.dsde.workbench.leonardo.dao.HostStatus.HostReady
+import org.broadinstitute.dsde.workbench.leonardo.dns.RuntimeDnsCache
+import org.broadinstitute.dsde.workbench.leonardo.model.LeoException
+import org.broadinstitute.dsde.workbench.leonardo.{AppContext, CloudContext, RuntimeName}
+import org.broadinstitute.dsde.workbench.model.TraceId
+import org.broadinstitute.dsde.workbench.openTelemetry.OpenTelemetryMetrics
+import org.http4s._
+import org.http4s.circe.CirceEntityDecoder._
+import org.http4s.client.Client
+import org.http4s.client.dsl.Http4sClientDsl
+import org.http4s.client.middleware.{Retry, RetryPolicy}
+import org.typelevel.log4cats.Logger
+
+import scala.concurrent.duration._
+import scala.util.control.NoStackTrace
+
+class HttpCromwellDAO[F[_]](val runtimeDnsCache: RuntimeDnsCache[F], httpClient: Client[F], samDAO: SamDAO[F])(implicit
```

Comment:
I don't think we need `RuntimeDnsCache` anymore...becuz cromwell on Azure's proxy path will be using relay

Reply:
I think we should plan for `HttpCromwellDAO` to support GKE and AKS.

…However `RuntimeDnsCache` is for VMs, this should probably use `KubernetesDnsCache` for GKE.

…And I agree, for Azure we probably don't need a DNS cache at all. Can we just build the relay hybrid connection
URL directly (how is that done in other places)?

Reply:
yea you should be able to build the relay url directly..for runtime, we persist the relay url in DB, and the hybridconnection name is created using the runtime name

Reply:
OK.. for apps we are also creating the hybridConnection as the app name: https://github.com/DataBiosphere/leonardo/blob/develop/http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AKSInterpreter.scala#L111

But we're not persisting it in the DB. Perhaps we need to add a call like [this](https://github.com/DataBiosphere/leonardo/blob/develop/http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala#L231) in AKSInterpreter..

Reply:
yea, I think that makes sense...all apps on the same cluster will have the same relay namespace..